index.js

```diff
@@ -1,7 +1,9 @@
 const utils = require('loader-utils');
 
 function performReplacement(source, options) {
-    if (options.search && options.replace) {
+    let searchDefined = (typeof options.search !== "undefined") && (options.search !== null);
+    let replaceDefined = (typeof options.replace !== "undefined") && (options.replace !== null);
```

Comment:
[q] I believe `options.search && options.replace` will catch more edge cases.

Reply:
your suggestion it the original implementation. In the case that option.replace it the empty string "" this won't work. The empty string is counted as false. Therefore I have changed to the above implementation.

Reply:
Good catch, in this case I will suggest `Boolean(options.search)` and `Boolean(options.replace)`. Thanks for making those changes.

Reply:
After we merge, I'm going to create a new version with your changes and then you should be able to use that version on your project.

Reply:
I executed the following code in the browser console:

`let a = Boolean('');
console.log(a);`

And it outputs "false". This means with your suggested change a replace with empty string is still not possible.

Reply:
Ahh okay, thats what you want to do:

In this case let's make it implicit then.

What do you think about:
```
const searchDefined = Boolean(options.search) || options.search === ''
```

this will make sure all edge cases are still in place and add empty string. Sorry about the misunderstanding.

Please change them to `const` as those values aren't going to change.